mula/scheduler/connectors/services/katalogus.py

```diff
@@ -115,3 +155,26 @@ def get_normalizers_by_org_id_and_type(self, organisation_id: str, normalizer_ty
         except dict_utils.ExpiredError:
             self._flush_organisations_normalizer_type_cache()
             return dict_utils.deep_get(self.organisations_normalizer_type_cache, [organisation_id, normalizer_type])
+
+    def get_new_boefjes_by_org_id(self, organisation_id: str) -> List[Plugin]:
+        # Get the enabled boefjes for the organisation from katalogus
+        plugins = self.get_plugins_by_organisation(organisation_id)
+        enabled_boefjes = {
+            plugin.id: plugin for plugin in plugins if plugin.enabled is True and plugin.type == "boefje"
+        }
+
+        # Check if there are new boefjes
+        new_boefjes = []
+        for boefje_id, boefje in enabled_boefjes.items():
+            if boefje_id in self.organisations_new_boefjes_cache.get(organisation_id, {}):
```

Comment:
Would this not see each boefje as a new boefje if the cache has been cleaned? This would result in a large query to Octopoes for all active boefje's their cumulative OOI-Type set instead of just the actual change.
Im sot sure a Cache is the right way to do this. Maybe a set compare to the last run is better?

Reply:
It sets the `enabled_boefjes` as the current collection of boefjes (checked at that moment). It should at a new call of `get_new_boefjes_by_org_id` check the difference since the last call to the method.